src/submission/views.py

```diff
@@ -131,14 +131,17 @@ def submit_funding(request, article_id):
             article.save()
             return redirect(reverse('submit_review', kwargs={'article_id': article_id}))
 
-        funder = models.Funder(
-            name=request.POST.get('funder_name', default=''),
-            fundref_id=request.POST.get('funder_doi', default=''),
-            funding_id=request.POST.get('grant_number', default=''),
+        funder_form = forms.Funder(
+            {
+                'name': request.POST.get('funder_name', None),
+                'fundref_id': request.POST.get('funder_doi', None),
+                'funding_id': request.POST.get('grant_number', None)
+            }
         )
 
-        funder.save()
-        article.funders.add(funder)
+        if funder_form.is_valid():
+            funder = funder_form.save()
+            article.funders.add(funder)
```

Comment:
Could the funder form take an article argument and add this as part of the save process? It would make it easier to reuse the form.

Reply:
Good catch. Added some commits that move the saving routine inside the form